tests/safety/test_hyundai.py

```diff
@@ -239,6 +244,77 @@ def _gas_msg(self, gas):
     values = {"CR_Vcu_AccPedDep_Pos": gas}
     return self.packer.make_can_msg_panda("E_EMS11", 0, values, fix_checksum=checksum)
 
+class TestHyundaiLongitudinalSafety(TestHyundaiSafety):
+  TX_MSGS = [[832, 0], [1265, 0], [1157, 0], [1056, 0], [1057, 0], [1290, 0], [905, 0], [1186, 0], [909, 0], [1155, 0], [2000, 0]]
+
+  def setUp(self):
+    self.packer = CANPackerPanda("hyundai_kia_generic")
+    self.safety = libpandasafety_py.libpandasafety
+    self.safety.set_safety_hooks(Panda.SAFETY_HYUNDAI, Panda.FLAG_HYUNDAI_LONG)
+    self.safety.init_tests()
+
+  # override these tests from PandaSafetyTest, hyundai longitudinal uses button enable
+  def test_disable_control_allowed_from_cruise(self):
+    pass
+
+  def test_enable_control_allowed_from_cruise(self):
+    pass
+
+  def test_cruise_engaged_prev(self):
+    pass
+
+  def _pcm_status_msg(self, enable):
+    raise NotImplementedError
+
+  def _button_msg(self, buttons):
+    values = {"CF_Clu_CruiseSwState": buttons}
+    return self.packer.make_can_msg_panda("CLU11", 0, values)
+
+  def _send_accel_msg(self, accel):
+    values = {
+      "aReqRaw": accel,
+      "aReqValue": accel,
+    }
+    return self.packer.make_can_msg_panda("SCC12", 0, values)
+
+  def _send_fca11_msg(self, idx=0, aeb_req=False, aeb_decel=0):
+    values = {
+      "CR_FCA_Alive": ((-((idx % 0xF) + 2) % 4) << 2) + 1,
+      "Supplemental_Counter": idx % 0xF,
+      "FCA_Status": 2,
+      "CR_VSM_DecCmd": aeb_decel,
+      "CF_VSM_DecCmdAct": int(aeb_req),
+      "FCA_CmdAct": int(aeb_req),
+    }
+    return self.packer.make_can_msg_panda("FCA11", 0, values)
+
+  def test_no_aeb(self):
+    self.assertTrue(self._tx(self._send_fca11_msg()))
+    self.assertFalse(self._tx(self._send_fca11_msg(aeb_req=True)))
+    self.assertFalse(self._tx(self._send_fca11_msg(aeb_decel=1.0)))
+
+  def test_resume_button(self):
```

Comment:
button tests should be combined or a new test should be added to check the whole button signal space